src/viam/examples/modules/complex/gizmo/api.hpp

```diff
@@ -44,6 +43,13 @@ class Gizmo : public Component {
     explicit Gizmo(std::string name);
 };
 
+template <>
+struct API::api_map<Gizmo> {
+    static API api() {
+        return {"viam", "component", "gizmo"};
```

Comment:
Just FYI, you don't need to implement these in the header. You can declare it here but define it out of line, and I recommend doing so to keep these strings out of the headers (they were in the `.cpp` before).

Reply:
Ahh okay great. For some reason I thought template specializations had to be in the `.hpp` file, but will move these implementations into the `.cpp` files.

Reply:
Oh! So not only is this cleaner and keeping strings outside of headers, but somehow it seems to have solved that bizarre bug I was encountering. I'd be lying if I said I understood _why_ that was the case but I'm happy to see resolution at least.